Cannot Build on x86_64

the error i get with build.sh 

injected.m:152:11: warning: format string is not a string literal (potentially insecure) [-Wformat-security]
    NSLog(baseDirectory);
          ^~~~~~~~~~~~~
injected.m:152:11: note: treat the string as an argument to avoid this
    NSLog(baseDirectory);
          ^
          @"%@", 
injected.m:221:1: warning: non-void function does not return a value in all control paths [-Wreturn-type]
}
^
2 warnings generated.

this is what it shows now:

injected.m:152:11: warning: format string is not a string literal (potentially insecure) [-Wformat-security]
    NSLog(baseDirectory);
          ^~~~~~~~~~~~~
injected.m:152:11: note: treat the string as an argument to avoid this
    NSLog(baseDirectory);
          ^
          @"%@", 
injected.m:221:1: warning: non-void function does not return a value in all control paths [-Wreturn-type]
}
^
2 warnings generated.
Undefined symbols for architecture x86_64:
  "_NSHomeDirectory", referenced from:
      _try_decrypt_epub in injected-e9f96f.o
      _discoverEPUB in injected-e9f96f.o
  "_NSLog", referenced from:
      _get_sinf in injected-e9f96f.o
      _try_decrypt in injected-e9f96f.o
      _write_file in injected-e9f96f.o
      _write_file in injected-e9f96f.o
      _write_file in injected-e9f96f.o
      _make_base_epub_dir in injected-e9f96f.o
      _try_decrypt_epub in injected-e9f96f.o
      _try_decrypt_epub in injected-e9f96f.o
      _try_decrypt_epub in injected-e9f96f.o
      _try_decrypt_epub in injected-e9f96f.o
      _try_decrypt_epub in injected-e9f96f.o
      _try_decrypt_epub in injected-e9f96f.o
      _try_decrypt_epub in injected-e9f96f.o
      _try_decrypt_epub in injected-e9f96f.o
      ...
  "_OBJC_CLASS_$_NSConstantArray", referenced from:
       in injected-e9f96f.o
       in injected-e9f96f.o
  "_OBJC_CLASS_$_NSData", referenced from:
       in injected-e9f96f.o
  "_OBJC_CLASS_$_NSFileManager", referenced from:
       in injected-e9f96f.o
  "_OBJC_CLASS_$_NSString", referenced from:
       in injected-e9f96f.o
  "___CFConstantStringClassReference", referenced from:
       in injected-e9f96f.o
       in injected-e9f96f.o
       in injected-e9f96f.o
       in injected-e9f96f.o
       in injected-e9f96f.o
       in injected-e9f96f.o
       in injected-e9f96f.o
      ...
  "_objc_alloc", referenced from:
      _get_sinf in injected-e9f96f.o
  "_objc_enumerationMutation", referenced from:
      _try_decrypt_epub in injected-e9f96f.o
  "_objc_getClass", referenced from:
      _try_decrypt in injected-e9f96f.o
  "_objc_msgSend", referenced from:
      _get_sinf in injected-e9f96f.o
      _get_sinf in injected-e9f96f.o
      _get_sinf in injected-e9f96f.o
      _get_sinf in injected-e9f96f.o
      _get_sinf in injected-e9f96f.o
      _get_sinf in injected-e9f96f.o
      _get_sinf in injected-e9f96f.o
      ...
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

now it shows this: 

injected.m:152:11: warning: format string is not a string literal (potentially insecure) [-Wformat-security]
    NSLog(baseDirectory);
          ^~~~~~~~~~~~~
injected.m:152:11: note: treat the string as an argument to avoid this
    NSLog(baseDirectory);
          ^
          @"%@", 
injected.m:221:1: warning: non-void function does not return a value in all control paths [-Wreturn-type]
}
^
2 warnings generated.
 "/Library/Developer/CommandLineTools/usr/bin/ld" -demangle -lto_library /Library/Developer/CommandLineTools/usr/lib/libLTO.dylib -no_deduplicate -dynamic -dylib -arch x86_64 -platform_version macos 14.0.0 14.4 -syslibroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -o injected.dylib -L/usr/local/lib -lobjc /var/folders/qh/dc0_lqxj2791lvymgc_gpctr0000gn/T/injected-8dca0c.o -lSystem /Library/Developer/CommandLineTools/usr/lib/clang/15.0.0/lib/darwin/libclang_rt.osx.a
Undefined symbols for architecture x86_64:
  "_NSHomeDirectory", referenced from:
      _try_decrypt_epub in injected-8dca0c.o
      _discoverEPUB in injected-8dca0c.o
  "_NSLog", referenced from:
      _get_sinf in injected-8dca0c.o
      _try_decrypt in injected-8dca0c.o
      _write_file in injected-8dca0c.o
      _write_file in injected-8dca0c.o
      _write_file in injected-8dca0c.o
      _make_base_epub_dir in injected-8dca0c.o
      _try_decrypt_epub in injected-8dca0c.o
      _try_decrypt_epub in injected-8dca0c.o
      _try_decrypt_epub in injected-8dca0c.o
      _try_decrypt_epub in injected-8dca0c.o
      _try_decrypt_epub in injected-8dca0c.o
      _try_decrypt_epub in injected-8dca0c.o
      _try_decrypt_epub in injected-8dca0c.o
      _try_decrypt_epub in injected-8dca0c.o
      ...
  "_OBJC_CLASS_$_NSConstantArray", referenced from:
       in injected-8dca0c.o
       in injected-8dca0c.o
  "_OBJC_CLASS_$_NSData", referenced from:
       in injected-8dca0c.o
  "_OBJC_CLASS_$_NSFileManager", referenced from:
       in injected-8dca0c.o
  "_OBJC_CLASS_$_NSString", referenced from:
       in injected-8dca0c.o
  "___CFConstantStringClassReference", referenced from:
       in injected-8dca0c.o
       in injected-8dca0c.o
       in injected-8dca0c.o
       in injected-8dca0c.o
       in injected-8dca0c.o
       in injected-8dca0c.o
       in injected-8dca0c.o
      ...
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)


also it opens the tmp folder with the epub there, but its unreadable in Calibre.

--- a/build.sh
+++ b/build.sh
@@ -1,8 +1,19 @@
-if [[ $(nvram boot-args) != *"arm64e_preview_abi"* ]]; then
-	echo "arm64e preview ABI is not enabled."
-	echo "enable it with: sudo nvram boot-args=-arm64e_preview_abi"
-	echo "and then restart your mac."
-	exit 1
-fi
+arches=("arm64e" "x86_64")
 
-clang -dynamiclib -o injected.dylib injected.m -arch arm64e -framework Foundation
+select arch in "${arches[@]}";
+do
+	# Validate the user's input
+    if [[ "$REPLY" =~ ^[0-9]+$ ]] && [ "$REPLY" -ge 0 ] && [ "$REPLY" -le ${#arches[@]} ]; then
+    	if [ "$arch" == "arm64e" ]; then
+			if [[ $(nvram boot-args) != *"arm64e_preview_abi"* ]]; then
+				echo "arm64e preview ABI is not enabled."
+				echo "enable it with: sudo nvram boot-args=-arm64e_preview_abi"
+				echo "and then restart your mac."
+				exit 1
+			fi
+		fi
+		
+		clang -dynamiclib -o injected.dylib injected.m -arch "$arch" -framework Foundation
+	fi
+	break
+done

--- a/decrypt.sh
+++ b/decrypt.sh
@@ -25,14 +25,20 @@ mkdir ./decrypted_books
 
 epubFiles=()
 itemNames=()
+index=1
 
 for file in "$BOOKS_EPUB_DIR"/*.epub
 do
-    itemName=$(xmllint --xpath "/plist/dict/key[text()='itemName']/following-sibling::*[1]/text()" "$file/iTunesMetadata.plist" 2>/dev/null)
+    itemName=$(defaults read "$file/iTunesMetadata" itemName)
     if [[ -z "$itemName" ]]; then
         echo "Failed to extract itemName from $file"
         continue
     fi
+    # If there are more than 21 files, select does not list all the options
+    if [ "$index" -ge 22 ]; then
+        echo "$index) $itemName"
+    fi
+    index=$((index+1))
     epubFiles+=("$file")
     itemNames+=("$itemName")
 done

--- a/injected.m
+++ b/injected.m
@@ -52,8 +52,14 @@ NSData* try_decrypt(NSData *sinfData, NSString *path) {
     // Call the DRM symbol...
     NSData* result = [cls pK0gFZ9QOdm17E9p9cpP:path sinfData:sinfData refetch:&refetch error:&error];
     if (result == nil) {
-        printf("Failed to decrypt: %s\n", [[error localizedDescription] UTF8String]);
-        return nil;
+        // Error code -42010 means there is no DRM protection to decrypt
+        if (error.code == -42010) {
+            result = [NSData dataWithContentsOfFile:path  options:0 error:nil];
+        }
+        else {
+            printf("Failed to decrypt: %s\n", [[error localizedDescription] UTF8String]);
+            return nil;
+        }
     }
     NSLog(@"Decrypted: %@", result);
 

